test/e2e/ytsaurus_controller_test.go

```diff
@@ -1097,7 +1097,7 @@ func getSimpleUpdateScenario(namespace, newImage string) func(ctx context.Contex
 		DeferCleanup(deleteYtsaurus, ytsaurus)
 		name := types.NamespacedName{Name: ytsaurus.GetName(), Namespace: namespace}
 		deployAndCheck(ytsaurus, namespace)
-		oldGeneration := ytsaurus.ObjectMeta.Generation
+		oldGeneration := ytsaurus.Status.ObservedGeneration
```

Comment:
I think we better test that `ytsaurus.ObjectMeta.Generation` (not ObservedGeneration) of object we have after applying spec for update (I think `k8sClient.Update(ctx, ytsaurus)` should set new value for Generation field in the struct) is the same that we have in `ytsaurus.Status.ObservedGeneration` after the update is finished. This is the scenario we plan to use that field.

Reply:
Done